tracer/src/Datadog.Trace/Debugger/Expressions/ProbeProcessor.cs

```diff
@@ -0,0 +1,268 @@
+                        {
+                            return true;
+                        }
+
+                        break;
+                    case MethodState.EntryEnd:
+                    case MethodState.ExitEnd:
+                    case MethodState.EndLine:
+                    case MethodState.EndLineAsync:
+                    case MethodState.ExitEndAsync:
+                        try
+                        {
+                            if (snapshotCreator.CaptureBehaviour == CaptureBehaviour.NoCapture)
+                            {
+                                return true;
+                            }
+
+                            snapshotCreator.AddScopeMember(info.Name, info.Type, info.Value, info.MemberKind);
+                            evaluationResult = GetOrCreateEvaluator(snapshotCreator.MethodScopeMembers).Evaluate();
+
+                            if (evaluationResult.Condition is false && (evaluationResult.Errors == null || evaluationResult.Errors.Count == 0))
+                            {
+                                // if the expression evaluated to false or the evaluation failed
+                                snapshotCreator.CaptureBehaviour = CaptureBehaviour.NoCapture;
+                                return true;
+                            }
+
+                            if (evaluationResult.Metric.HasValue)
+                            {
+                                LiveDebugger.Instance.SendMetrics();
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            // if the evaluation failed stop capturing
+                            Log.Error(e, "Failed to evaluate expression for probe: " + ProbeInfo.ProbeId);
+                            snapshotCreator.CaptureBehaviour = CaptureBehaviour.NoCapture;
+                            return true;
+                        }
+
+                        break;
+                    case MethodState.LogLocal:
+                    case MethodState.LogArg:
+                    case MethodState.LogException:
+                        if (snapshotCreator.CaptureBehaviour == CaptureBehaviour.NoCapture)
+                        {
+                            // there is a condition that should evaluate at exit phase and we are at entry phase
+                            return true;
+                        }
+
+                        snapshotCreator.AddScopeMember(info.Name, info.Type, info.Value, info.MemberKind);
+
+                        if (snapshotCreator.CaptureBehaviour == CaptureBehaviour.Delayed)
+                        {
+                            return true;
+                        }
+
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+
+                ProcessCapture(ref info, snapshotCreator, ref evaluationResult);
```

Comment:
Only one specific `case` in the preceding `switch/case` assigns a value in `evaluationResult`, but you send it to `ProcessCapture` in any case. Is it by design?